### ISSUE TITLE
Allow constructing a git_rebase with a custom list of git_rebase_operation

### ISSUE THREAD

**Issue by joshtriplett** [libgit2/libgit2#3795]
Currently, libgit2 only supports construci a `git_rebase` from a trio of annotated commits, or by opening an existing rebase in progress.  I'd like to construct a `git_rebase` representing a `git rebase -i` operation; to do that, I'd like to have a way to construct a `git_rebase` with a custom series of `git_rebase_operation` structures.

**Comment by carlosmn** [comment]
Yes, it's something we'd want to do at some point, but it requires designing the interface for that and what a consumer of the library would need. If you'd like to do that, that'd be great.

**Comment by tiennou** [comment]
As an alternative, maybe providing a callback that gets called at each step would be easier ? I've tried to design something like `git_rebase_init_interactive`, but since the list of operations is generated from annotated commits, meaning consumers would have to duplicate the revwalking in [rebase_init_operations](https://github.com/libgit2/libgit2/blob/master/src/rebase.c#L584).

How custom would be your series of operations ? Because this wouldn't handle *really custom* series of operations that don't make sense in the current `branch`/`upstream`/`onto` paradigm. Or was it more standard, like the usual `git rebase` workflow of fixups, skips and reorders ?

**Comment by joshtriplett** [comment]
@tiennou I mostly would want to see support for all the existing `rebase -i` operations, rather than just `rebase`'s "take x..y and rebase it on top of z".
